README.md

```diff
@@ -1,16 +1,131 @@
-# Title
+# Learning from Your Questions
 
 ## Learning Goals
 
--SWBAT 1
--SWBAT 2
+- Identify effective methods of formulating questions
+- Identify community standards for writing help requests
 
 ## Introduction
 
-## SWBAT 1
+Before we get the right answer, we need to know how to ask the right question.
+Figuring out how to ask the right question is a process of clarifying exactly
+what we want the result to be as well as taking stock of all the information we
+currently have. The more thought we put into asking questions, the better chance
+we give ourselves of getting the right answer—either from others, or by
+ourselves.
 
-## SWBAT 2
+## Identify Effective Methods of Formulating Questions
+
+Sometimes, it's clear to us what we don't know and what we need to ask for. But
+often it's not so clear. Often, we don't know what we don't know and that makes
+it difficult to formulate a question to ask about it. In these not-so-clear
+cases, we can start with a process of determining what we _do_ know. Eventually,
+this process will lead us to the right question.
+
+Here's some steps to take when you're questions:
+
+1. __Focus on small section of code.__ Don’t keep scanning several files for a
+bug. Reducing the amount of material to sort through will allow you to isolate
+the issues faster and easier.
+
+2. __State your inputs.__ Double-check what data is getting pulled into your
+program. Is your code reading the correct file? Were the values entered
+correctly? If an input is wrong, it could throw off all of your code.
+
+3. __State your outputs.__ Similarly, double-check what data is getting pushed
+out of your program. If you're not getting out what you want, you might be able
+to trace it backwards to the source of the problem.
+
+4. __State your expectation.__ In order to determine whether your outputs are
+right or wrong, you need to compare what you're getting to what you expected to
+get.
+
+5. __State your theory for what should have happened.__ You might have an idea
+what you wanted to get as an result, but how did you get that idea? Double-check
+your understanding of how the code is working. Did you miss anything that would
+explain a different result?
+
+6. __Check the docs to make sure your implementation works like you think it
+did.__ As you're exploring your theory, take a look at any relevant reference
+documentation. All developers have been in a position where they _thought_
+something worked a certain way, but in fact did not. Double-check all the
+available documentation for whichever tool or technique you're using so that you
+can eliminate that as a contributing factor to the current issue.
+
+7. __Test the implementation in IRB.__ There's no substitute for running bits of
```

Comment:
putting this brick after intro to programming basic ruby or ruby methods would set students up to understand IRB